src/util/eventBus.ts

```diff
@@ -0,0 +1,30 @@
+import { EventData } from '@/redux/events/types';
+
+type EventHandler = () => void;
+type EventType = EventData['Event'];
+
+/**
+ * This class can be used to subscribe to OpenSpace events in components. For now, it does
+ * not get any data with the event, but rather just a notification that it happened
+ */
+class EventBus {
+  private listeners: Partial<Record<EventType, EventHandler[]>> = {};
+
+  on(event: EventType, handler: EventHandler) {
+    if (!this.listeners[event]) {
+      this.listeners[event] = [];
+    }
+    this.listeners[event].push(handler);
+  }
+
+  off(event: EventType, handler: EventHandler) {
+    this.listeners[event] = this.listeners[event]?.filter((h) => h !== handler) || [];
+  }
+
+  // @TODO (anden88 2025-10-09): Do we want to pass the event payload to the handler?
+  emit(event: EventType) {
+    this.listeners[event]?.forEach((handler) => handler());
+  }
+}
+
+export const eventBus = new EventBus();
```

Comment:
I'm a little confused with this class. It seems to me that it is valid for all the events we have in the event system and the events middleware. but it is only used for the `AssetLoadingFinished`? I have two concerns:
1) Why is it written so generally if it is only used for the one type?
2) Why is it breaking the pattern from the other ways we handle the events? Are there perfomance drawbacks to have the listeners as a state in redux? Is there a way we can make this code be more in line with how we handle the other events? Can we use redux actions instead of the functions for example?